File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
       unordered_map<int,int> mp;
       for(int ele:nums){
            mp[ele]++;
       }
       for(const auto& ele : mp){
            if(ele.second >=2) return true;
       } 
       return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
       
       Arrays.sort(nums);
       System.out.print(nums);

       for(int i = 1; i < nums.length; i++) {
         if(nums[i] == nums[i-1]) {
            return true;
         }
       }

       return false;

        // int n = nums.length;

        // for(int i = 0; i < n; i++) {
        //     for(int j = i + 1; j < n; j++) {
        //         if(nums[i] == nums[j]) {
        //             return true;
        //         }
        //     }
        // }
        // return false;
    }
}